images/linux/scripts/SoftwareReport/SoftwareReport.CachedTools.psm1

```diff
@@ -33,7 +33,21 @@ function Get-ToolcacheBoostVersions {
     if (-not (Test-Path $toolcachePath)) {
         return @()
     }
-    return Get-ChildItem $toolcachePath -Name | Sort-Object { [Version]$_ }
+
+    $ToolInstances = Get-CachedToolInstances -Name $Name
```

Comment:
`Get-CachedToolInstances` doesn't exist on Linux.

I think we should do something like below
```pwsh
$boostVersions = Get-ChildItem $toolcachePath -Name | Sort-Object { [Version]$_ }
$toolInstances = $boostVersions | ForEach-Object {
  $VersionEnvVar = $_.replace(".", "_")
  return @{
    Version = $_
    Architecture = "x64"
    "Environment Variable" = "BOOST_ROOT_${VersionEnvVar}"
    
  }
}
$Content = $ToolInstances | New-MDTable -Columns ([ordered]@{
  Version = "left";
  Architecture = "left";
  "Environment Variable" = "left"
})

return Build-MarkdownElement -Head $Name -Content $Content
```

Reply:
Thanks @maxim-lobanov. I use a Linux machine so it is hard for me to test this and I am not good at Powershell scripting.

Reply:
`Build-MarkdownElement` doesn't exist as a helper function for Linux report generation. It could be replaced with
```
$markdown = New-MDHeader $Name -Level 4
$markdown += New-MDParagraph -Lines $Content

return $markdown
```

Reply:
@varunagrawal, feel free to create an issue and we will take care about this change

Reply:
@varunagrawal , you can create an issue in our repo to implement this change in docs, and our team will do that if it is not convenient for you to test these changes and spend time on Powershell scripting.

Reply:
I spent some time understanding the basic ideas behind Powershell scripting and I think I've gotten the hang of it.

Perhaps a better issue would be to refactor the code base to allow for the helper functions to be accessible across the scripts? Lots of reinventing-the-wheel happening here.

Reply:
We investigated the possibility to create a common helpers code base for all images but didn't implement that due to a one reason:
as you can see, we are running validation with "azp run <image>" if the image-related code changed. If we used common helpers - we'd have to run all images validation on each common code update and it would take a significant amount of time. 

So here we will consider adding the same helpers functions to Linux code base as we have for Windows to make development easier.